src/language_server_protocol.rs

```diff
@@ -2041,6 +2042,36 @@ impl LanguageClient {
         Ok(Value::Null)
     }
 
+    pub fn progress(&self, params: &Value) -> Fallible<()> {
+        let params = ProgressParams::deserialize(params)?;
+        let message = match params.value {
+            ProgressParamsValue::WorkDone(wd) => match wd {
+                WorkDoneProgress::Begin(r) => {
+                    Some(format!("{} {}", r.title, r.message.unwrap_or_default()))
+                }
+                WorkDoneProgress::Report(r) => r.message,
+                // WorkDoneProgress::End has no value, so we return Done, otherwise the previous
+                // message would be left in screen and it would appear as if it didn't ever finish.
+                WorkDoneProgress::End(_) => Some("Done".into()),
+            },
+        };
+
+        if message.is_none() {
+            return Ok(());
+        }
+
+        let token = match params.token {
+            // number is a not a particularly useful token to report to the user, so we just use
+            // INFO instead.
+            NumberOrString::Number(_) => "INFO".to_string(),
+            NumberOrString::String(s) => s,
+        };
+
+        let message = format!("{}: {}", token, message.unwrap());
```

Comment:
Nit: I'd suggest update `message.unwrap()` => `message.upwrap_or_default()`.

Though message itself has already been checked previously, `None` might slip in in future changes. `unwrap`  would halt the whole program, which is very unfortunate.

I'm actually considering adding a clippy rule to warn/error about `.unwrap`.